lib/rspec/rails/example/view_example_group.rb

```diff
@@ -161,6 +163,12 @@ def _include_controller_helpers
           controller.controller_path = _controller_path
           controller.request.path_parameters[:controller] = _controller_path
           controller.request.path_parameters[:action]     = _inferred_action unless _inferred_action =~ /^_/
+          controller.request.path = ViewPathBuilder.new(::Rails.application.routes).path_for(controller.request.path_parameters)
+          ViewSpecMethods.add_to(::ActionView::TestCase::TestController)
+        end
+
+        after do
+          ViewSpecMethods.remove_from(::ActionView::TestCase::TestController)
```

Comment:
My only concern is adding a new API to the spec. If for some reason my view does look at `params`, say `params[:id]`, and looks at the path (I know not something that should be done, but it still could happen), the `id` value will need to get set twice. That seems a bit odd to me.

I thought about suggesting we override `render` in the spec, but then I realized that my standard debugging technique wouldn't necessarily work there:

```ruby
p controller.request.path
render
```

Instead the inspection would need to occur after `render`. Which may not be an issue unless `render` causes an exception in which case inspecting the path wouldn't be possible without more hackery.

I'm honestly a bit torn between the extra API and having to shim around `params`.